controller/program/tests/tests/protocol_fee/v2/withdraw_protocol_fees.rs

```diff
@@ -0,0 +1,535 @@
+                withdraw_to_aft,
+                &token_acc_bal_diff_changed(withdraw_to_bef, expected_minted as i128),
+            );
+
+            assert_eq!(
+                inf_mint_aft.supply() - inf_mint_bef.supply(),
+                expected_minted
+            );
+
+            assert_eq!(pool_aft.protocol_fee_lamports, 0);
+        }
+        Some(e) => {
+            assert_jiminy_prog_err(&result.unwrap_err(), e);
+        }
+    }
+}
+
+#[test]
+fn withdraw_protocol_fees_v2_correct_basic() {
+    const INF_MINT_SUPPLY: u64 = 10_000_000_000_000;
+    const WITHDRAW_TO_BALANCE: u64 = 50;
+    const PROTOCOL_FEE_LAMPORTS: u64 = 1_000_000_000;
+    const TOTAL_SOL_VALUE: u64 = 100_000_000_000_000;
+
+    // 69 + to avoid colliding with system prog
+    let [ben, wt] = core::array::from_fn(|i| [69 + u8::try_from(i).unwrap(); 32]);
+
+    let pool = PoolStateV2FtaVals {
+        addrs: PoolStateV2Addrs::default()
+            .with_protocol_fee_beneficiary(ben)
+            .with_lp_token_mint(INF_MINT_ID),
+        u64s: PoolStateV2U64s::default()
+            .with_protocol_fee_lamports(PROTOCOL_FEE_LAMPORTS)
+            .with_total_sol_value(TOTAL_SOL_VALUE),
+        ..Default::default()
+    }
+    .into_pool_state_v2();
+
+    let keys = NewWithdrawProtocolFeesV2IxAccsBuilder::start()
+        .with_pool_state(POOL_STATE_ID)
+        .with_beneficiary(ben)
+        .with_withdraw_to(wt)
+        .with_inf_mint(INF_MINT_ID)
+        .with_token_program(TOKENKEG_PROGRAM)
+        .build();
+
+    withdraw_protocol_fees_v2_test(
+        withdraw_protocol_fees_v2_ix(&keys),
+        &withdraw_protocol_fees_v2_test_accs(&keys, pool, INF_MINT_SUPPLY, WITHDRAW_TO_BALANCE),
+        Option::<ProgramError>::None,
+    );
+}
+
+fn to_inp(
+    (keys, pool, inf_mint_supply, withdraw_to_balance): (
+        WithdrawProtocolFeesV2IxKeysOwned,
+        PoolStateV2,
+        u64,
+        u64,
+    ),
+) -> (Instruction, AccountMap) {
+    (
+        withdraw_protocol_fees_v2_ix(&keys),
+        withdraw_protocol_fees_v2_test_accs(&keys, pool, inf_mint_supply, withdraw_to_balance),
+    )
+}
+
+fn correct_strat() -> impl Strategy<Value = (Instruction, AccountMap)> {
+    (
+        any_normal_pk(),
+        any_pool_state_v2(PoolStateV2FtaStrat {
+            u8_bools: pool_state_v2_u8_bools_normal_strat(),
+            addrs: PoolStateV2Addrs::default().with_lp_token_mint(Some(Just(INF_MINT_ID).boxed())),
+            u64s: PoolStateV2U64s::default()
+                .with_protocol_fee_lamports(Some((0..=u64::MAX / 1000).boxed()))
+                .with_total_sol_value(Some((1..=u64::MAX).boxed()))
+                .with_last_release_slot(Some(Just(0).boxed())),
+            ..Default::default()
+        })
+        .prop_filter("pool must be solvent for LPs", |ps| {
+            ps.withheld_lamports
+                .saturating_add(ps.protocol_fee_lamports)
+                <= ps.total_sol_value
+        }),
```

Comment:
should probably copy-pasta `pool_sv_lamports_invar_strat` over to `test-utils` and use it to create a `Strategy<Value = PoolStateV2U64s>` similar to what we did with `PoolStateBools::normal()`